src/libraries/System.Text.RegularExpressions/tests/RegexCultureTests.cs

```diff
@@ -119,5 +141,137 @@ Regex[] Create(string input, CultureInfo info, RegexOptions additional)
+                yield return new object[] { "(?i:[^IJKLM]I)", options, current, "ii\u0130i\u0131ab", "" };
+
+                // Expected answers in the invariant culture
+                yield return new object[] { "(?i:I)", options, invariant, "xy\u0131ab", "" };
+                yield return new object[] { "(?i:iI+)", options, invariant, "abcIIIxyz", "III" };
+                yield return new object[] { "(?i:iI+)", options, invariant, "abc\u0130\u0131Ixyz", "" };
+
+                // Expected answers in the Turkish culture
+                yield return new object[] { "(?i:I)", options, turkish, "xy\u0131ab", "\u0131" };
+                yield return new object[] { "(?i:iI+)", options, turkish, "abcIIIxyz", "" };
+                yield return new object[] { "(?i:iI+)", options, turkish, "abcIi\u0130xyz", "" };
+                yield return new object[] { "(?i:iI+)", options, turkish, "abcI\u0130ixyz", "" };
+                yield return new object[] { "(?i:[^IJKLM]I)", options, turkish, "ii\u0130i\u0131ab", "i\u0131" };
+
+                if (options != RegexOptions.None && options != RegexOptions.Compiled)
+                {
+                    // Otherwise the following tests fail (in disjoint cases), see one extracted active issue below
+
+                    // Expected answers in the invariant culture
+                    yield return new object[] { "(?i:iI+)", options, invariant, "abcIi\u0130xyz", "Ii" };               // <-- failing for None, Compiled
+                    yield return new object[] { "(?i:iI+)", options, invariant, "abcI\u0130ixyz", "" };                 // <-- failing for Compiled
+                    yield return new object[] { "(?i:iI+)", options, invariant, "abc\u0130IIxyz", "II" };               // <-- failing for Compiled
+                    yield return new object[] { "(?i:iI+)", options, invariant, "abc\u0130Iixyz", "Ii" };               // <-- failing for Compiled
+                    yield return new object[] { "(?i:[^IJKLM]I)", options, invariant, "ii\u0130i\u0131ab", "\u0130i" }; // <-- failing for None, Compiled
+
+                    // Expected answers in the Turkish culture
+                    yield return new object[] { "(?i:iI+)", options, turkish, "abc\u0130IIxyz", "\u0130II" };           // <-- failing for None, Compiled
+                    yield return new object[] { "(?i:iI+)", options, turkish, "abc\u0130\u0131Ixyz", "\u0130\u0131I" }; // <-- failing for None, Compiled
+                    yield return new object[] { "(?i:iI+)", options, turkish, "abc\u0130Iixyz", "\u0130I" };            // <-- failing for None, Compiled
+                }
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(Match_In_Different_Cultures_TestData))]
+        public void Match_In_Different_Cultures(string pattern, RegexOptions options, CultureInfo culture, string input, string match_expected)
+        {
+            Regex r = Create(pattern, culture, options)[0];
```

Comment:
this can just new up the regex itself here?

Reply:
right, it was just convenient to use the existing Create method for passing the culture, else I need to set and then undo setting the thread culture. But I can add another Create1 method that does not create 2 regexes.

Reply:
Or just have one CreateRegex method that takes culture & options and takes care of the culture reset. Tests that need both call it twice once with Compiled and once without. Would probably be clearer in the test.

Reply:
I added `RegexHelpers.CreateRegexInCulture(pattern, options, culture)`